lib/fs.js

```diff
@@ -2479,3 +2479,457 @@ Object.defineProperty(fs, 'SyncWriteStream', {
   set: internalUtil.deprecate((val) => { SyncWriteStream = val; },
                               'fs.SyncWriteStream is deprecated.', 'DEP0061')
 });
+
+
+// The Promises API
+
+async function* readEverything(fd) {
+  let eof = false;
+  while (!eof) {
+    const buffer = Buffer.allocUnsafe(kReadFileBufferLength);
+    const { length } =
+      await fs.promises.read(fd, buffer, 0, buffer.length);
+    if (length < buffer.length)
+      eof = true;
+    yield buffer.slice(0, length);
+  }
+}
+
+
+async function writeEverything(fd, buffer, offset, length, position) {
+  const { written } =
+    await fs.promises.write(fd, buffer, offset, length, position);
+  if (written !== length) {
+    offset += written;
+    length -= written;
+    if (position !== null)
+      position += written;
+    return writeEverything(fd, buffer, offset, length, position);
```

Comment:
If the buffer is big enough, it could lead to a stack overflow. Can this be done as a loop?

Reply:
Given that it returns a Promise, does it overflow the stack? :-) 

That said, I'm already planning to rework this function. This was a temporary impl